src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7320,6 +7342,69 @@ private struct NetResource
             [MarshalAs(UnmanagedType.LPWStr)]
             public string Provider;
         }
+
+        #region InodeTracker
+        /// <summary>
+        /// Tracks visited files/directories by caching their device IDs and inodes.
+        /// </summary>
+        private class InodeTracker
+        {
+            private HashSet<(UInt64, UInt64)> _visitations;
+
+            /// <summary>
+            /// Construct a new InodeTracker with an initial path
+            /// </summary>
+            internal InodeTracker(string path)
+            {
+                _visitations = new HashSet<(UInt64, UInt64)>();
+
+                if (InternalSymbolicLinkLinkCodeMethods.GetInodeData(path, out (UInt64, UInt64) inodeData))
+                {
+                    _visitations.Add(inodeData);
+                }
+            }
+
+            /// <summary>
+            /// Determine if a path has been visited. Optionally mark the path as
+            /// having been visited if it has not already
+            /// </summary>
+            /// <param name="path">
+            /// Path to the file system item to be checked.
+            /// </param>
+            /// <param name="visitIfNotAlready">
+            /// Flag to indicate whether the path should be marked as visited if
+            /// it has not already been visited.
```

Comment:
Normally I would say no. In this case the existence of the parameter and its name give a clue about the side effect of marking the path as visited.